src/dataflow/source/kafka.rs

```diff
@@ -356,6 +356,14 @@ where
                     let partition = message.partition;
                     let offset = message.offset + 1;
 
+                    // Offsets are guaranteed to be 1) monotonically increasing *unless* there is
+                    // a network issue or a new partition added, at which point the consumer may
+                    // start processing the topic from the beginning, or we may see duplicate offsets
```

Comment:
does this mean we should expect to see things stall when new partitions are added?

Reply:
No, when a new partition is added, we'll see an offset that is smaller than the last offset that we recorded for this partition, and we'll seek to that last offer.